cyberdrop_dl/managers/download_manager.py

```diff
@@ -69,9 +70,12 @@ async def basic_auth(self, username, password) -> str:
         token = b64encode(f"{username}:{password}".encode('utf-8')).decode("ascii")
         return f'Basic {token}'
 
-    async def check_free_space(self) -> bool:
+    async def check_free_space(self, folder: Path = None) -> bool:
         """Checks if there is enough free space on the drive to continue operating"""
-        free_space = shutil.disk_usage(self.manager.path_manager.download_dir.parent).free
+        if folder is None:
+            free_space = disk_usage(self.manager.path_manager.download_dir.drive).free
+        else:
+            free_space = disk_usage(folder).free
```

Comment:
Yeah, `.drive` won't work on unix. It always returns the root. Also, i was wrong about `download_dir`. It's created at `downloader.startup()` which is called inside `crawler.startup()`. 

https://github.com/jbsparrow/CyberDropDownloader/blob/b4dc0021f041cd2574d6b21194d6f18928cd8e76/cyberdrop_dl/downloader/downloader.py#L127

That's before any download and space check.  Using `self.manager.path_manager.download_dir` works just fine. 
The problem is with the parameter `folder`. I was able to fix it with this:

```suggestion
        if folder is None:
            free_space = disk_usage(self.manager.path_manager.download_dir).free
        else:
            folder = folder.resolve()
            while not folder.is_dir() and folder.parents:
                folder = folder.parent   
            # check if we reached an anchor (root) that does not exists, ex: disconnected USB drive
            if not folder.is_dir(): 
                return False
            free_space = disk_usage(folder).free
```
Basically, if `folder` does not exists, traverse it until you find a parent that does and check free space on it

Its works but now there is another minor issue. The download fails but is does not show in the UI summary, only in the logs. UI shows 0 files failed or skipped and no download errors.

**Logs**

[downloader.log](https://github.com/user-attachments/files/17092217/downloader.log)

![UI_screenshot](https://github.com/user-attachments/assets/33fbbaef-7ffb-4dfb-ae10-36673a08ba74)

Reply:
Resolved, the files were being added as skipped to the download progress bar but not for the download stats manager.